multiqc/modules/metadmg/metadmg.py

```diff
@@ -0,0 +1,244 @@
+""" MultiQC module to plot output from metadmg """
+
+
+import logging
+import pandas as pd
+
+from multiqc import config
+from multiqc.modules.base_module import BaseMultiqcModule, ModuleNoSamplesFound
+from multiqc.plots import bargraph, linegraph
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """
+    To allow reading gzip archives, run with `ignore_images: false`
+    in the config, e.g.:
+    ```
+    multiqc . --cl-config 'ignore_images: false'
+    ```
+    """
+
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="metaDMG",
+            anchor="metadmg",
+            href="https://github.com/metaDMG-dev/metaDMG-cpp",
+            info="Taxonomic classification of reads and estimation of damage rates in ancient DNA data",
+            doi="10.1101/2022.12.06.519264",
+        )
+
+        # Read config
+        self.rank = getattr(config, "metadmg", {}).get("rank", "genus")
+        self.n_taxa = getattr(config, "metadmg", {}).get("n_taxa", 20)
+        self.sort_by = getattr(config, "metadmg", {}).get("sort_by", "nreads")
+        self.df_filter = getattr(config, "metadmg", {}).get("df_filter", "nreads > 5")
+        self.index = getattr(config, "metadmg", {}).get("index", "name")
+
+        # Read files
+        self.metadmg_data = self.metadmg_read_data()
+
+        # Filter to strip out ignored sample names
+        self.metadmg_data = self.ignore_samples(self.metadmg_data)
+
+        if len(self.metadmg_data) == 0:
+            raise ModuleNoSamplesFound
+        else:
+            log.info(f"Found {len(self.metadmg_data.keys())} samples: {self.metadmg_data.keys()}")
+        # Write parsed report data to a file
+        self.write_data_file(self.metadmg_data, "multiqc_metadmg")
+
+        # Add version
+        self.add_software_version(None)
+
+        # Get all taxa
+        self.taxa = sorted(list(set([str(taxon) for s_name, s_data in self.metadmg_data.items() for v_name, v_data in s_data.items() for taxon in v_data])))
+
+        # Plots section
+        self.barplot_section()
+        #self.dfitplot_samples_section()
+        self.dfitplot_taxa_section()
+
+    def metadmg_read_data(self):
+        # Number of positions used
+        n_positions = list()
+
+        df_metadmg = dict()
+        for f in self.find_log_files("metadmg/stat", filecontents=False):
+            f["s_name"] = f["s_name"][:-5]
+            # Read DF
+            df_metadmg[f["s_name"]] = pd.read_table(
+                f["fn"],
+                sep="\t",
+                index_col="taxid",
+                comment="#",
+            )
+            # Add data
+            self.add_data_source(f, f["s_name"])
+
+        for f in self.find_log_files("metadmg/dfit", filecontents=False):
+            f["s_name"] = f["s_name"][:-5]
+            # Read DF
+            df_dfit = pd.read_table(
```

Comment:
Is it going to parse `*.dfit.gz` as well?

Reply:
Yes, that was the idea. I thought of pandas since it makes it much easier to read csv/tsv files (even if they are gzipped).